### PR TITLE
7428 - Fix on showColumn not working properly with frozen columns

### LINKED ISSUE #7428: Datagrid: Hiding then restoring columns does not work properly when there are frozen columns
**Describe the bug**
Hiding then restoring columns does not work properly when there are frozen columns

**To Reproduce**
Could be replicated here https://main-enterprise.demo.design.infor.com/components/datagrid/example-frozen-columns.html

Initial datagrid
image

Column 'Activity' is hidden successfully after uncheck in 'Personalize Columns', then after checking again
image

The header for 'Activity' is now in the first frozen column, then 'Quantity' is now the header for activity, it seems the headers are moved to the left as the last column doesn't have a heading.

**Expected behavior**
Columns to be aligned properly after restoring column.

**Version**

- ids-enterprise: 4.82.0

### LINKED ISSUE #7541: DataGrid: Frozen column header are mismatching/order incorrect, while reselecting the personalize columns
**Describe the bug**
Frozen Columns are mismatching when reselecting the personalize columns in datagrid

**To Reproduce**
Could be replicated here https://main-enterprise.demo.design.infor.com/components/datagrid/example-frozen-columns.html
Select the frozen columns from datagrid.
Initial datagrid when selecting personalize columns.
image

Now deselect Activity column and deselect the Quantity column.
Now again select the Activity column and select the Quantity column.

The result shown as: 
image

The Frozen column headers are overriding the personalize columns header.

**Expected behavior**
The headers should not move from personalize columns.

**Version**

### PR DESCRIPTION
**Pull request by yohannahbautista**
**Explain the _details_ for making this change. What existing problem does the pull request solve?**

showColumn was inserting another column in the datagrid for the given column, under the assumption that it's a hidden checkbox column. Added an additional check when assigning headerNodeCheckbox so that the Activity header in the example won't be inserted on the first column.

**Related github/jira issue (required)**:

Closes #7428 
Closes #7541

**Steps necessary to review your pull request (required)**:

- Pull this branch, build and run the app
- Go to: http://localhost:4000/components/datagrid/example-frozen-columns.html
- Click on Personalize Columns
- Deselect and select Activity so that the column will disappear and appear
- Columns should render correctly (Activity should not be pushed on the first column and values should be correct)

**Included in this Pull Request**:
- [ ] A note to the change log.

### CODE REVIEW

File: docs/CHANGELOG.md [hunk 1]
```diff
@@ -13,6 +13,7 @@
 - `[Button]` Adjusted alignment for popupmenu icon buttons. ([#7408](https://github.com/infor-design/enterprise/issues/7408))
 - `[Charts]` Improved the positioning of chart legend. ([#7452](https://github.com/infor-design/enterprise/issues/7452))
 - `[Datagrid]` Clear `rowstatus` in tree node for clearRowError to work correctly. ([#6033](https://github.com/infor-design/enterprise/issues/6033))
+- `[Datagrid]` Fix on `showColumn` not working properly with frozen columns. ([#7428](https://github.com/infor-design/enterprise/issues/7428))
```
Comment (ericangeles) [inline docs/CHANGELOG.md:16]: Fixed an issue where `showColumn` was not functioning correctly with frozen columns.

**Review by tmcconechy — DISMISSED** [review]

**Review by glenlieorillo — CHANGES_REQUESTED** [review]
Bug already fixed. However, if you deselect the 'Activity' column the border between 'Product Name' and 'Quantity' is missing.


https://github.com/infor-design/enterprise/assets/102151352/1847a84c-44e7-4b2b-9955-e9cf03ceec4e

image

**Comment by tmcconechy** [comment]
@yohannahbautista can you take a look at that point by @glenlieorillo

**Review by ericangeles — APPROVED** [review]

**Review by glenlieorillo — DISMISSED** [review]
QA Approved.


https://github.com/infor-design/enterprise/assets/102151352/b8fa3063-b99b-42d1-81bc-84de668f60ed

For Clarification:
@tmcconechy 
Upon deselecting all columns, there is a big gap between 'Product Name' and 'Button' columns. Kindly confirm if this is the correct behavior. Thanks.

image

**Review by jbrcna — DISMISSED** [review]

**Comment by tmcconechy** [comment]
@glenlieorillo its probably "ok" as its an edge case and the column widths are fixed and that is the size. So unsure what else can be done. At least for future consideration

**Comment by glenlieorillo** [comment]
> @glenlieorillo its probably "ok" as its an edge case and the column widths are fixed and that is the size. So unsure what else can be done. At least for future consideration

Thanks, @tmcconechy.

**Pull request merged**